maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -0,0 +1,192 @@
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.Disposable;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.Initializable;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.InitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service to configure the {@link OpenTelemetry} instance.
+ *
+ * <p>Mimic the <a
+ * href="https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure">OpenTelemetry
+ * SDK Autoconfigure</a> that can't be used due to class loading issues when declaring the Maven
+ * OpenTelemetry extension using the pom.xml {@code <extension>} declaration: the OpenTelemetry SDK
+ * Autoconfigure registers a JVM shutdown hook that is incompatible with the fact that Maven
+ * extensions are unloaded before the JVM shuts down, causing {@link NoClassDefFoundError} when the
+ * shutdown hook is invoked.
+ */
+@Component(role = OpenTelemetrySdkService.class, hint = "opentelemetry-service")
+public final class OpenTelemetrySdkService implements Initializable, Disposable {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  @Requirement private RuntimeInformation runtimeInformation;
+
+  private OpenTelemetrySdk openTelemetrySdk;
+
+  private Tracer tracer;
+
+  private SpanExporter spanExporter;
+
+  @Override
+  public synchronized void dispose() {
+    logger.debug("OpenTelemetry: dispose OpenTelemetrySdkService...");
+    if (this.openTelemetrySdk != null) {
+      logger.debug("OpenTelemetry: Shutdown SDK Trace Provider...");
+      long before = System.currentTimeMillis();
+      final CompletableResultCode sdkProviderShutdown =
+          this.openTelemetrySdk.getSdkTracerProvider().shutdown();
+      sdkProviderShutdown.join(10, TimeUnit.SECONDS);
+      if (sdkProviderShutdown.isSuccess()) {
+        logger.debug(
+            "OpenTelemetry: SDK Trace Provider shutdown in "
+                + (System.currentTimeMillis() - before)
+                + " ms");
+      } else {
+        logger.warn(
+            "OpenTelemetry: Failure to shutdown SDK Trace Provider in "
+                + (System.currentTimeMillis() - before)
+                + " ms, done: "
+                + sdkProviderShutdown.isDone()
+                + " success: "
+                + sdkProviderShutdown.isSuccess());
+      }
+      // reset GlobalOpenTelemetry as we have just shutdown its TraceProvider
+      GlobalOpenTelemetry.resetForTest();
+      this.openTelemetrySdk = null;
+    }
+    logger.debug("OpenTelemetry: OpenTelemetrySdkService disposed");
+  }
+
+  /** TODO add support for `OTEL_EXPORTER_OTLP_CERTIFICATE` */
+  @Override
+  public void initialize() throws InitializationException {
+    logger.debug("OpenTelemetry: initialize OpenTelemetrySdkService...");
+    // OTEL_EXPORTER_OTLP_ENDPOINT
```

Comment:
Is there a reason not to use the autoconfigure module by the way?

Reply:
I explained the reason in the javadoc of the OpenTelemetrySdkService class, it due to the fact that the Autoconfigure uses a JVM shutdown hook to close the Trace Provider which is incompatible with the lifecycle of a Maven `<extension>` that is unloaded from the classloader way before the JVM shuts down, causing a `NoClassDefFoundError` in the shutdown hook

See 
https://github.com/open-telemetry/opentelemetry-java/blob/v1.6.0/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/TracerProviderConfiguration.java#L58
```java
Runtime.getRuntime().addShutdownHook(new Thread(tracerProvider::close));
```

Reply:
I opened an issue to track this #82